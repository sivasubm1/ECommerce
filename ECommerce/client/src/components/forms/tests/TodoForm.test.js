import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import TodoForm from "../TodoForm";
import { test } from "@jest/globals";

afterEach(() => {
  cleanup();
});

test("should render non-completed TodoForm component ", () => {
  const todo = { id: 1, title: "send status email", completed: false };

  render(<TodoForm todo={todo} />);
  const todoElement = screen.getByTestId("todo-1");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("send status email");
});

test("should render completed TodoForm component ", () => {
  const todo = { id: 2, title: "enter timesheet", completed: true };

  render(<TodoForm todo={todo} />);
  const todoElement = screen.getByTestId("todo-2");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("enter timesheet");
});

test("matches snapshot", () => {
  const todo = { id: 1, title: "send status email", completed: false };
  const tree = renderer.create(<TodoForm todo={todo} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("matches snapshot1", () => {
  const todo = { id: 2, title: "enter timesheet", completed: true };
  const tree = renderer.create(<TodoForm todo={todo} />).toJSON();
  expect(tree).toMatchSnapshot();
});
