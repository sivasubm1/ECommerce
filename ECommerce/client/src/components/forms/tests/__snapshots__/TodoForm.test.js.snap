// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <h1>
    send status email
  </h1>
</div>
`;

exports[`matches snapshot1 1`] = `
<div
  data-testid="todo-2"
>
  <strike>
    <h1>
      enter timesheet
    </h1>
  </strike>
</div>
`;
